ts_algorithms/sirt.py

```diff
@@ -41,5 +45,7 @@ def sirt(A, y, num_iterations):
         A.T(y_tmp, out=x_tmp)
         x_tmp *= C
         x_cur -= x_tmp
+        if min_constraint or max_constraint:
```

Comment:
0 also evaluates to False, so in the common case of min_constraint == 0 and max_constraint == None, this code wouldn't clamp. You should check for the None value by using "is None" or in this case "min_constraint is not None"

https://stackoverflow.com/questions/3965104/not-none-test-in-python

Reply:
Good point, I haven't thought about it.